### PR TITLE
feat: Support Cloud Logging in Java via Logback

### PR DESCRIPTION
**Pull request by NimJay**
### Background

* Fixes https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/issues/67.
* Logback is one of the most popular logging Java framework. It's a successor to Log4j (also a popular Java logging framework).
* According to [cloud.google.com](https://cloud.google.com/logging/docs/setup/java):
> You can write logs to Cloud Logging from Java applications by using 
> the [Logback appender](https://cloud.google.com/logging/docs/setup/java#logback_appender) 
> or a [java.util.logging handler](https://cloud.google.com/logging/docs/setup/java#jul_handler), 
> or by using the Cloud Logging library for Java directly. 
* An "appender" is a Logback concept. Appenders can be used to direct logs to services such as Cloud Logging.

### Summary of changes
* These changes add support for Google Cloud Logging by using Logback and the Google Cloud "Logback appender".
* This change uses something called Spring Boot "Profiles" to switch between local logging and Cloud Logging. See [docs about Profiles](https://docs.spring.io/spring-boot/reference/features/profiles.html).
    * The `default` Profile will use Cloud Logging.
    * The `locallogging` Profile (which is a custom profile I've introduced) will use local logging.
    * For some `mvn` commands, the Profile will be set to `default` by default. See [docs](https://docs.spring.io/spring-framework/reference/core/beans/environment.html#beans-definition-profiles-default).

### CODE REVIEW

File: app/java/Procfile
Comment (NimJay) [inline app/java/Procfile:2]: To make the data `process`-ing job output logs locally (instead of Cloud Logging), we need to replace `default` with `locallogging`:
```
java -jar target/cloud-client-api-0.0.0-SNAPSHOT-processingjob.jar -Dspring-boot.run.profiles=locallogging
```

However, this is not documented anywhere currently.
@glasnt, what are you thoughts on introducing a `make` command for local execution of the `process` job? Alternatively, I can document the above state here: https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/tree/main/app#application

File: app/java/src/main/resources/logback-spring.xml
Comment (NimJay) [inline app/java/src/main/resources/logback-spring.xml:1]: I renamed the `logback.xml` file to `logback-spring.xml` because of [guidance here](https://docs.spring.io/spring-boot/docs/2.1.1.RELEASE/reference/html/boot-features-logging.html#boot-features-custom-log-configuration):
> When possible, we recommend that you use the -spring variants for your logging configuration (for example, logback-spring.xml rather than logback.xml). If you use standard configuration locations, Spring cannot completely control log initialization.

**Comment by NimJay** [comment]
The failing `lint` Check will be addressed via https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/86/files/1a607580b18a04ea446ad9c49e041c726e44c8b6#diff-4b122024a3a28ded65da76a2f1bface1f3a27328374438d1298d25585fe0603b.

File: infra/variables.tf
Comment (NimJay) [inline infra/variables.tf:21]: Reminder to change this back:
```suggestion
  default     = "0.5.0" // x-release-please-version
```

Comment (glasnt) [reply infra/variables.tf:21]: I'm going to update this to instead alter the application_image local in main.tf. Changing this value here causes linting issues.

File: infra/main.tf
Comment (glasnt) [inline infra/main.tf:20]: Reminder to revert this before merging

File: app/java/Procfile
Comment (glasnt) [reply app/java/Procfile:2]: Having a `make process` that is for local execution of the job makes sense. It would be the companion to `make dev`. I'll create an issue/PR to add this. (This new `make` target shouldn't block this PR, it's a development helper only.)

File: app/java/src/main/resources/logback-spring.xml
Comment (glasnt) [inline app/java/src/main/resources/logback-spring.xml:44]: This logging level is exceptionally noisy. Trying to `make dev` flooded me with condition evaluation reports on startup, and thymeleaf processing messages on every page load. 

Suggest changing this to info, to match the logging level in the default profile
```suggestion
    <root level="info">
```

File: app/java/Procfile
Comment (NimJay) [reply app/java/Procfile:2]: I just approved https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/92.

File: app/java/src/main/resources/logback-spring.xml
Comment (NimJay) [reply app/java/src/main/resources/logback-spring.xml]: Good call. Applied suggestion.

File: infra/main.tf
Comment (NimJay) [reply infra/main.tf:20]:
```suggestion
  application_image = "us-docker.pkg.dev/hsa-public/containers/cloud-client-api/${lower(var.language)}:${var.image_version}"
```

**Review by glasnt — APPROVED** [review]

**Pull request merged**
